packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -730,3 +711,148 @@ class _RenderChildOverflowBox extends RenderAligningShiftedBox {
     }
   }
 }
+
+// Generate a FloatingActionButtonThemeData that represents
+// the M2 default values. This was generated by hand from the
+// previous hand coded defaults for M2. It uses get method overrides
+// instead of properties to avoid computing values that we may not
+// need upfront.
+class _M2Defaults extends FloatingActionButtonThemeData {
+  _M2Defaults(BuildContext context, this.type, this.hasChild)
+      : theme = Theme.of(context),
+        colors = Theme.of(context).colorScheme;
+
+  final _FloatingActionButtonType type;
+  final bool hasChild;
+  final ThemeData theme;
+  final ColorScheme colors;
+
+  bool get isExtended => type == _FloatingActionButtonType.extended;
+  bool get isLarge => type == _FloatingActionButtonType.large;
```

Comment:
Since its a private class, it probably doesn't matter, but should all these properties by private too?

Reply:
Good point, will fix.